modules/react/button/lib/BaseButton.tsx

```diff
@@ -21,6 +21,10 @@ import {CSSObject} from '@emotion/styled';
 import {CanvasSystemIcon} from '@workday/design-assets-types';
 
 export interface ButtonContainerProps extends BoxProps, GrowthBehavior {
+  /**
+   * Override default colors of a button. The default will depend on the button type
+   * @default {undefined}
```

Comment:
Is it customary for us to use JSDoc to specify a `@default` of `undefined`? I only see it in one other place in our source code (SecondaryButton `variant` has `@default undefined`).

Reply:
The defaults parser will try to evaluate defaults. In this case, it is evaluating an object that takes in parameters, so it is not static. The `undefined` overrides this behavior.

The default parser can be more sophisticated and bail if it detects this.

For example:

```tsx
{
  default: {
    background: string,
    border: string,
    icon: string,
    label: string
  }
}
```

That's not very useful. If the `get*ButtonColors` had a return type of `ButtonColors`, the default is a symbol linking to `ButtonColors`. Default rendering can be difficult because it uses the same recursive function as types do even if the purpose is different.

Reply:
This is what it looks like: 
<img width="820" alt="image" src="https://user-images.githubusercontent.com/338257/221323329-0a693385-87ee-4778-9339-b9f59e393463.png">

Reply:
I've added some logic when determining defaults. Right now, a valid default is anything matching a literal. We can expand this later if we feel literals are too limiting, but this prevents us from needing to add `undefined` or have these large defaults that don't tell us useful information.